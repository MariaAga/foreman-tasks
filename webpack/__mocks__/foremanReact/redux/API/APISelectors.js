export const selectAPIResponse = (state, key) => ({
  data: {
    text: 'some-data',
    key,
  },
});

export const selectAPIStatus = (state, key) => 'PENDING';
export const selectAPIError = (state, key) => ({ error: `${key} ERRROR` });
