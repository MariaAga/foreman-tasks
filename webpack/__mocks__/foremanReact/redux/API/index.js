export const API = {
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  patch: jest.fn(),
};

export const get = data => ({ type: 'some-type', ...data });
