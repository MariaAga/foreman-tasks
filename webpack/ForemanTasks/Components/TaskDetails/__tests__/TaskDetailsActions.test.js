import { testActionSnapshotWithFixtures } from '@theforeman/test';
import API from 'foremanReact/API';
import {
  taskReloadStop,
  taskReloadStart,
  fetchTaskDetails,
  cancelStep,
} from '../TaskDetailsActions';

jest.mock('foremanReact/API');

API.get.mockImplementation(async () => ({ data: 'some-data' }));

const fixtures = {
  'should start reload': () => taskReloadStart(1),
  'should stop reload': () => taskReloadStop(2),
  'should fetch-task-details and success': () => fetchTaskDetails(),
  'should cancelStep and success': () => cancelStep('task-id', 'step-id'),
};

describe('TaskDetails - Actions', () =>
  testActionSnapshotWithFixtures(fixtures));
