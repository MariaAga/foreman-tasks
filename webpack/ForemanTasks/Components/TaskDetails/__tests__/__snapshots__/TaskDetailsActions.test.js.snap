// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetails - Actions should fetch-task-details and success 1`] = `
Array [
  Array [
    Object {
      "type": "FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": "some-data",
      "type": "FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS",
    },
  ],
]
`;

exports[`TaskDetails - Actions should start reload 1`] = `
Object {
  "payload": Object {
    "id": undefined,
    "refetchTaskDetails": undefined,
    "timeoutId": 1,
  },
  "type": "FOREMAN_TASK_DETAILS_START_PULLING",
}
`;

exports[`TaskDetails - Actions should stop reload 1`] = `
Object {
  "payload": Object {
    "timeoutId": 2,
  },
  "type": "FOREMAN_TASK_DETAILS_STOP_PULLING",
}
`;
